### ISSUE TITLE
implement GPG directly in k9

### ISSUE THREAD

**Issue by webratte** [thunderbird/thunderbird-android#6085]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same feature request.
- [X] I will describe the problem with as much detail as possible.
- [X] This issue only contains a request for one single feature, **not** multiple (related) features.

### App version

all versions

### Problem you are trying to solve

It wuold be great to implement GPG directly into k9 without the need to use openkeychain. Because it's not developed anymore.

### Suggested solution

Maybe a similar solution as Delta Chat used:
https://github.com/deltachat/
https://github.com/deltachat/deltachat-android

**Comment by vanitasvitae** [comment]
If you need an OpenPGP library, I happen to be the author of [PGPainless](https://github.com/pgpainless/pgpainless) and I'd love to assist you :)

**Comment by ndalliard** [comment]
Any update on this? A [comment from Ryan Sipes](https://blog.thunderbird.net/2022/06/faq-thunderbird-mobile-and-k-9-mail/#comment-2251) from **2022** said that **this is on the list**.

**Comment by kewisch** [comment]
Hi folks, we're tracking this on Mozilla Connect, please see https://connect.mozilla.org/t5/ideas/inbuilt-pgp-system/idi-p/74739 for details and upvote if you'd like to see it. We regularly look at Connect, especially for roadmap planning.

I'm closing this issue to avoid tracking in two places at once, though it changes nothing about the importance of this issue.

Either way, we're monitoring the situation with OpenKeyChain. While it is currently not actively developed, they are still doing security maintenance. If there are announcements this would change then we'll take appropriate action.

**Comment by webratte** [comment]
Yes I know, you closed this issue, because you tracking it on a other place. But there comes also no comment abaut inbuild E2EE in Thunderbird-Android. 
So I ask here again. 

Will you implent PGP/GPG directly im Thunderbird-Android before OpenKeychain will die? 

Thanx

**Comment by rafaeltonholo** [comment]
> Yes I know, you closed this issue, because you tracking it on a other place. But there comes also no comment abaut inbuild E2EE in Thunderbird-Android. So I ask here again.
> 
> Will you implent PGP/GPG directly im Thunderbird-Android before OpenKeychain will die?
> 
> Thanx

Hello, thank you for reaching out.

Implementing PGP/GPG directly in Thunderbird is a significant undertaking, and unfortunately, we currently do not have the resources to manage it.

As @kewisch mentioned, we are keeping an eye on the situation. If there is a need, we will evaluate how we might proceed with this.

Thanks

**Comment by webratte** [comment]
Thank you for your reply. I understand your problem with the lack of resources. But it would indeed be a great shame if I could no longer read my emails on my smartphone at some point. I use my email provider's "post-encryption" feature. This means that every single one of my emails is encrypted using PGP. I would have a real problem if Thunderbird-Android could no longer handle PGP.

Nevertheless, I would like to thank the development team for their work so far. I really hope PGP doesn't lose focus 😁

**Comment by kewisch** [comment]
@webratte I'm curious which provider you use that offers this? We're considering something like this as well for Thunderbird Pro.

**Comment by webratte** [comment]
Take a look to www.mailbox.org.
This is a german Mailprovider. 

They also post-encrypt unencryptet (e.g with Thunderbird) sent mails.

**Comment by webratte** [comment]
@kewisch Will you now set more focus to PGP Integration? 😉
You see, without working PGP, Thunderbird-Android will become useless for me. I belive, I'm not the only one with this "problem".

**Comment by vanitasvitae** [comment]
BC now offers a high-level OpenPGP API similar to PGPainless: https://github.com/bcgit/bc-java/blob/main/pg/src/main/java/org/bouncycastle/openpgp/api/OpenPGPApi.java
I suggest you'd give it a try :)

I'm currently in the process of moving more and more complexity out of PGPainless back into BC.

**Comment by kewisch** [comment]
Thanks for the pointer @vanitasvitae! I see you also have a Kotlin version as well, which would be great for our quest to reduce java usage.

@webratte I believe right now OpenKeychain still works despite lower maintenance, so for the time being we're focusing on core stability. I totally understand the importance of this issue. When the urgency is higher we'll evaluate options as noted and seek out a solution.

**Comment by hayvan96** [comment]
At least would it be possible to have a different provider like the modern https://github.com/KeychainPGP/KeychainPGP precisely developed because of the state of openkeychain?

**Comment by webratte** [comment]
Maybe this could be a fast solution. 
But still I think the better way would be Integration in Thunderbird-Android (similar to Thunderbird-Desktop)
